### PR TITLE
SRE-1975 test: replace -hw with vm

### PR DESCRIPTION
**Pull request by daltonbohning**
All VM tests in master and 2.4 are tagged with "vm", so you "vm" instead of "-hw".

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -577,10 +577,7 @@ pipeline {
                                              'COMMIT_MESSAGE=' + cm.stripIndent()]) {
                                         cmp = commit.tag_template.replace('@commits.value@', commit.tags[0].value)
                                         cmp = cmp.replace('@stages.tag@', stage.tag)
-                                        // Useful for debugging since Jenkins'
-                                        // assert() is pretty lame
-                                        // println('assert(' + parseStageInfo()['test_tag'] + " == ${cmp})")
-                                        assert(parseStageInfo()['test_tag'] == cmp)
+                                        assert(parseStageInfo()['test_tag'] == cmp), parseStageInfo()['test_tag'] + ' != ' + cmp
```
Comment (daltonbohning) [inline Jenkinsfile:580]: Example:
https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/pipeline-lib/branches/PR-421/runs/8/nodes/81/log/?start=0
```
datamover,-hw != datamover,vm. Expression: assert(parseStageInfo()['test_tag'] == cmp), parseStageInfo()['test_tag'] + ' != ' + cmp
```

**Review by phender — DISMISSED** [review]

**Comment by daltonbohning** [comment]
The weekly-testing failed all stages
https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/ci-daos-stack-pipeline-lib-PR-421-weekly-testing/runs/1/nodes/79/log/?start=0
```
[2024-02-23T19:23:23.200Z] /var/lib/jenkins/jenkins-1/ci_node_wolf-137-141/workspace/peline-lib-PR-421-weekly-testing@tmp/durable-3d958fe4/script.sh: line 1: ci/functional/job_cleanup.sh: No such file or directory
```

**Review by phender — APPROVED** [review]

**Comment by daltonbohning** [comment]
@brianjmurrell This run passed with the Jenkinsfile self-reference to this PR, except the master build fault injection had an issue:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fpipeline-lib/detail/PR-421/11/pipeline

And the latest fails because it doesn't have the self-reference, so this PR would need to be force landed:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fpipeline-lib/detail/PR-421/12/pipeline

Are you in agreeance or should I run additional testing?

**Comment by brianjmurrell** [comment]
> @brianjmurrell This run passed with the Jenkinsfile self-reference to this PR, except the master build fault injection had an issue: [build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fpipeline-lib/detail/PR-421/11/pipeline](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fpipeline-lib/detail/PR-421/11/pipeline)

Do we know why this failed?

**Comment by daltonbohning** [comment]
> > @brianjmurrell This run passed with the Jenkinsfile self-reference to this PR, except the master build fault injection had an issue: [build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fpipeline-lib/detail/PR-421/11/pipeline](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fpipeline-lib/detail/PR-421/11/pipeline)
> 
> Do we know why this failed?

I don't, but it has been seen elsewhere in CI. And I can't imagine how this PR would affect that

**Comment by brianjmurrell** [comment]
> I don't, but it has been seen elsewhere in CI. And I can't imagine how this PR would affect that

Agreed.  It would just be nice to not have to question these sorts of things and force-land things against failing tests.  And then there is the cruft that failed test runs leave behind that needs to be cleaned up.

**Review by brianjmurrell — APPROVED** [review]

**Comment by brianjmurrell** [comment]
Please add @daos-stack/release-engineering to the reviewer list when you are ready for this to land (if you want somebody else to land this).

**Comment by daltonbohning** [comment]
> Please add @daos-stack/release-engineering to the reviewer list when you are ready for this to land (if you want somebody else to land this).

Who is generally gatekeeper for pipeline-lib? At least for daos we're usually not **_supposed_** to land our own PRs.

**Comment by brianjmurrell** [comment]
Landed.  Please delete the `ci-daos-stack-pipeline-lib-PR-421-master` branch that has been left around due to downstream test failures.  This is normal and expected as we want to leave those branches in-tact when there are failures so that the failures can be inspected.  Ideally a subsequent successful run, resolving the failures (assuming they weren't a known test failure that we are force-landing in despite of) would have cleaned up this branch for you.

**Comment by daltonbohning** [comment]
> Landed. Please delete the `ci-daos-stack-pipeline-lib-PR-421-master` branch that has been left around due to downstream test failures. This is normal and expected as we want to leave those branches in-tact when there are failures so that the failures can be inspected. Ideally a subsequent successful run, resolving the failures (assuming they weren't a known test failure that we are force-landing in despite of) would have cleaned up this branch for you.
> 
> Alternatively, perhaps one day we can create a GHA that can do this kind of (what should be unnecessary) garbage collection.

Thanks! Deleted. GHA can't be triggered "on merge", but can be triggered "on push". So we could have a GHA that triggers on push to master (which would have been a merge) to clean up stale branches. E.g.
```
on:
  push:
    branches:
      - master
```

**Comment by brianjmurrell** [comment]
> Thanks! Deleted. GHA can't be triggered "on merge", but can be triggered "on push". So we could have a GHA that triggers on push to master (which would have been a merge) to clean up stale branches. E.g.
> 
> ```
> on:
>   push:
>     branches:
>       - master
> ```

Very much indeed.  Just another thing on the pile though, and not near the top either.  Prioritizing is made even more difficult knowing that this is hack around other brokenness that if fixed would eliminate the need for it.

**Pull request merged**